kale/embed/_mpca_torch.py

```diff
@@ -0,0 +1,236 @@
+# =============================================================================
+# Author: Shuo Zhou, szhou20@sheffield.ac.uk
+#         Haiping Lu, h.lu@sheffield.ac.uk or hplu@ieee.org
+# =============================================================================
+
+"""Multilinear Principal Component Analysis (MPCA) implementation in Pytorch
+
+Reference:
+    Haiping Lu, K.N. Plataniotis, and A.N. Venetsanopoulos, "MPCA: Multilinear
+    Principal Component Analysis of Tensor Objects", IEEE Transactions on Neural
+    Networks, Vol. 19, No. 1, Page: 18-39, January 2008. For initial Matlab
+    implementation, please go to https://uk.mathworks.com/matlabcentral/fileexchange/26168.
+"""
+import logging
+
+# import pytorch_lightning as pl
+import warnings
+
+import tensorly as tl
+import torch
+import torch.nn as nn
+
+# from tensorly import fold, unfold
+from tensorly.tenalg import multi_mode_dot
+
+from .mpca import _check_dim_shape
+
+tl.set_backend("pytorch")
+
+
+class MPCA(nn.Module):
+    """MPCA implementation compatible with pytorch
+
+    Args:
+        var_ratio (float, optional): Percentage of variance explained
+            (between 0 and 1). Defaults to 0.97.
+        max_iter (int, optional): Maximum number of iteration. Defaults to 1.
+        mean_remove (bool, optional): Whether remove the mean of input data. Defaults ot True.
+            Set to False if the input data mean is zero already .
```

Comment:
white space before period

Reply:
In general, I prefer branches, in which case I could easily fix the problems, which will be much faster.